test/test_torrent.cpp

```diff
@@ -285,18 +295,15 @@ TORRENT_TEST(torrent)
 		file_storage fs;
 
 		fs.add_file("test_torrent_dir2/tmp1", 1024);
-		libtorrent::create_torrent t(fs, 128 * 1024, 6);
+		libtorrent::create_torrent t(fs, 1024, 6);
 
-		std::vector<char> piece(128 * 1024);
+		std::vector<char> piece(1024);
 		for (int i = 0; i < int(piece.size()); ++i)
 			piece[i] = (i % 26) + 'A';
 
 		// calculate the hash for all pieces
-		sha1_hash ph = hasher(&piece[0], piece.size()).final();
-		int num = t.num_pieces();
-		TEST_CHECK(t.num_pieces() > 0);
-		for (int i = 0; i < num; ++i)
-			t.set_hash(i, ph);
```

Comment:
I thought I reverted the changes to this test. As far as I can tell, the logic change does not require making these changes to the test (and it will be harder to merge in master). would you mind reverting these changes? (obviously leave in the new tests you added and the required changes like `wait_priority()`)